kolibri/plugins/coach/assets/src/state/actions/exam.js

```diff
@@ -49,51 +26,28 @@ function showExamsPage(store, classId) {
   store.dispatch('CORE_SET_PAGE_LOADING', true);
   store.dispatch('SET_PAGE_NAME', Constants.PageNames.EXAMS);
 
-  const currentClassPromise = ClassroomResource.getModel(classId).fetch();
-  const groupPromise = LearnerGroupResource.getCollection({ parent: classId }).fetch();
-  const channelPromise = ChannelResource.getCollection().fetch();
+  const resourceRequests = [
+    ClassroomResource.getModel(classId).fetch(),
+    LearnerGroupResource.getCollection({ parent: classId }).fetch(),
+    ChannelResource.getCollection().fetch(),
+    ExamResource.getCollectionForClass(classId).fetch(),
+  ];
 
-  ConditionalPromise.all([currentClassPromise, groupPromise, channelPromise]).only(
+  return ConditionalPromise.all(resourceRequests).only(
     CoreActions.samePageCheckGenerator(store),
-    ([currentClassModel, groupsCollection, channelsCollection]) => {
-      const currentClass = _classState(currentClassModel);
-      const currentClassGroups = _groupsState(groupsCollection);
-      const channels = _channelsState(channelsCollection);
-
-      const dummyExams = [{
-        id: '1',
-        title: 'UNIT 1 Exam',
-        active: false,
-        dateCreated: 'March 15, 2017 03:24:00',
-        visibility: { class: false, groups: [{ id: '1', name: 'groupA' }, { id: '2', name: 'groupA' }] },
-      },
-      {
-        id: '2',
-        title: 'UNIT 1 Quiz',
-        active: true,
-        dateCreated: 'March 21, 2017 03:24:00',
-        visibility: { class: false, groups: [{ id: '1', name: 'groupA' }] },
-      },
-      {
-        id: '3',
-        title: 'UNIT 2',
-        active: true,
-        dateCreated: 'March 22, 2017 03:24:00',
-        visibility: { class: true, groups: [{ id: '1', name: 'groupA' }, { id: '2', name: 'groupA' }] },
-      }];
-
+    ([classroom, learnerGroups, channels, exams]) => {
       const pageState = {
+        channels: _channelsState(channels),
         classId,
-        currentClass,
-        currentClassGroups,
-        channels,
-        exams: dummyExams,
+        currentClass: pickIdAndName(classroom),
+        currentClassGroups: learnerGroups.map(pickIdAndName),
+        exams,
```

Comment:
Exams are not transformed to proper format for pageState. Need to figure out the actual payload from api is,  and how to get it to format expected by view.